lib/streamlit/DeltaGenerator.py

```diff
@@ -651,7 +669,7 @@ def dataframe(self, _, data=None):
         def set_data_frame(delta):
             data_frame_proto.marshall_data_frame(data, delta.data_frame)
 
-        return self._enqueue_new_element_delta(set_data_frame)
+        return self._enqueue_new_element_delta(set_data_frame, width, height)
```

Comment:
Nice! You lucked out that `DeltaGenerator.data_frame` calls `_enqueue_new_element_delta` directly :smiley: . Other methods don't do that, and instead they have the `@_with_element` decorator that takes care of that.